### PR TITLE
Use StringEx to generate slugs

### PR DESCRIPTION
**Pull request by thomasleese**
See individual commits for more details, but this library seems to support more characters than the built in Rails transliteration and should avoid any issues we've been seeing with empty string outputs.

[Trello Card](https://trello.com/c/oedYOWVD/2247-updates-to-protected-food-drink-names-finder)

### CODE REVIEW

File: spec/models/protected_food_drink_name_spec.rb [hunk 1]
```diff
@@ -16,4 +16,12 @@
       expect(document.base_path).to eq("/protected-food-drink-names/sterea-ellada")
     end
   end
+
+  context "with Cryllic characters" do
```
Comment (benjamineskola) [inline spec/models/protected_food_drink_name_spec.rb]: "Cyrillic" :)

Comment (thomasleese) [reply spec/models/protected_food_drink_name_spec.rb]: Oops, I've pushed up a fix!

File: spec/models/protected_food_drink_name_spec.rb [hunk 2]
```diff
@@ -16,4 +16,12 @@
       expect(document.base_path).to eq("/protected-food-drink-names/sterea-ellada")
```
Comment (ChrisBAshton) [inline spec/models/protected_food_drink_name_spec.rb:16]: Am I right in assuming that the Stringex approach generates the same slug as the manual Greek transliteration approach?

**Review by ChrisBAshton — APPROVED** [review]
LGTM 👍  I wondered if this might change how English slugs are generated too, but the [README](https://github.com/rsl/stringex) looks pretty predictable, so I wouldn't have thought we'd run into that.

And it looks as though this code is only called on "first draft", so we don't have to worry about existing slugs, do we?

File: spec/models/protected_food_drink_name_spec.rb (hunk 2 above)
Comment (thomasleese) [reply spec/models/protected_food_drink_name_spec.rb:16]: Yeah, that's correct 👍

**Comment by thomasleese** [comment]
> And it looks as though this code is only called on "first draft", so we don't have to worry about existing slugs, do we?

That's right, this will only affect new editions.

**Comment by thomasleese** [comment]
This supersedes https://github.com/alphagov/specialist-publisher/pull/1771.

**Pull request merged**